flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_RequestsWeather.js

```diff
@@ -19,105 +12,73 @@ class CDUAocRequestsWeather {
         };
 
         mcdu.onRightInput[1] = (value) => {
-            if (value === FMCMainDisplay.clrValue) {
-                store["arpt2"] = "";
-            } else {
-                store["arpt2"] = value;
-            }
-            CDUAocRequestsWeather.ShowPage(mcdu, store);
-        };
-
-        mcdu.rightInputDelay[2] = () => {
-            return mcdu.getDelaySwitchPage();
+            mcdu.aocAirportList.set(1, value);
+            CDUAocRequestsWeather.ShowPage(mcdu, reqID, sendStatus);
         };
 
         mcdu.onRightInput[2] = (value) => {
-            if (value === FMCMainDisplay.clrValue) {
-                store["arpt3"] = "";
-            } else {
-                store["arpt3"] = value;
-            }
-            CDUAocRequestsWeather.ShowPage(mcdu, store);
-        };
-
-        mcdu.rightInputDelay[3] = () => {
-            return mcdu.getDelaySwitchPage();
+            mcdu.aocAirportList.set(2, value);
+            CDUAocRequestsWeather.ShowPage(mcdu, reqID, sendStatus);
         };
 
         mcdu.onRightInput[3] = (value) => {
-            if (value === FMCMainDisplay.clrValue) {
-                store["arpt4"] = "";
-            } else {
-                store["arpt4"] = value;
-            }
-            CDUAocRequestsWeather.ShowPage(mcdu, store);
+            mcdu.aocAirportList.set(3, value);
+            CDUAocRequestsWeather.ShowPage(mcdu, reqID, sendStatus);
         };
 
         mcdu.rightInputDelay[5] = () => {
             return mcdu.getDelaySwitchPage();
         };
 
         mcdu.onRightInput[5] = async () => {
-            const icaos = [store["arpt1"], store["arpt2"], store["arpt3"], store["arpt4"]];
-            if (icaos.every((current) => current === "")) {
+            const icaos = mcdu.aocAirportList.icaos;
+            if (icaos.length === 0) {
                 mcdu.addNewMessage(NXFictionalMessages.noAirportSpecified);
                 return;
             }
-            store["sendStatus"] = "QUEUED";
+            sendStatus = "QUEUED";
             updateView();
             const lines = [];
-            const newMessage = { "id": Date.now(), "type": reqTypes[store.reqID], "time": '00:00', "opened": null, "content": lines, };
+            const newMessage = { "id": Date.now(), "type": reqTypes[reqID], "time": '00:00', "opened": null, "content": lines, };
             mcdu.clearUserInput();
 
             const getInfo = async () => {
-                if (store.reqID == 0) {
-                    getMETAR(icaos, lines, store, updateView);
+                if (reqID === 0) {
+                    getMETAR(icaos, lines, sendStatus, updateView);
                 } else {
-                    getTAF(icaos, lines, store, updateView);
+                    getTAF(icaos, lines, sendStatus, updateView);
                 }
             };
 
             getInfo().then(() => {
-                store["sendStatus"] = "SENT";
+                sendStatus = "SENT";
```

Comment:
Can't we remove setting `sendStatus` in `getMETAR` and `getTAF` if it's already set here?

Reply:
intellij actually marks these two as unused xD

Reply:
Sorry for the nitpicking but then let's remove the parameters in the method as well and I'm happy :D

Reply:
Do you mean this parameters? `reqID` and `_sendStatus` from `static ShowPage(mcdu, reqID = 0, _sendStatus = "")`?

Reply:
Exactly, thanks!